src/screens/Authentication/Signin.tsx

```diff
@@ -1,41 +1,77 @@
-import React from "react";
-import { StyleSheet, Text, View, Image } from "react-native";
-import { useAuthentication } from "../../utils/hooks/useAuthentication";
-import { Pressable } from "react-native";
-import styles from "./styles";
-import { getAuth, signOut } from "firebase/auth";
+import React, { useRef } from 'react';
+import { StyleSheet, Text, View, Image } from 'react-native';
+import { useAuthentication } from '../../utils/hooks/useAuthentication';
+import { Pressable } from 'react-native';
+import styles from './styles';
+import { phoneGetConfirmation, confirmCode } from '../../firebase/auth';
+import { FirebaseRecaptchaVerifierModal } from 'expo-firebase-recaptcha';
+import { firebaseApp } from '../../firebase/config';
+import { getUser, addUser } from '../../firebase/firestore/user';
+// import { getAuth, signOut } from "firebase/auth";
 
-const auth = getAuth();
+// const auth = getAuth();
 
 const logo = require('../../assets/favicon.png');
 
 const SigninScreen = ({ navigation }: any) => {
   const { user } = useAuthentication();
+  const recaptchaVerifier = useRef(null);
 
   return (
+    <View style={styles.logocontainer}>
+      <Image source={logo} style={styles.logo} />
+      <View style={styles.container}>
+        <Text style={styles.signintext1}>
+          Welcome to the Chinese Newcomers Service Center job portal!
+        </Text>
+        <Text style={styles.signintext2}>Let's get your account set up.</Text>
 
-    <View style={styles.logocontainer} >
-    <Image source={logo} style={styles.logo}/>
-    <View style={styles.container}>
-      
-      <Text style={styles.signintext1}>
-        Welcome to the Chinese Newcomers Service Center job portal!</Text>
-      <Text style={styles.signintext2}>Let's get your account set up.</Text>
-     
-      <Pressable
-        style={styles.signinbutton}
-        onPress={() => navigation.navigate("Home")}
-      >
-        <Text style={styles.signintext3}> SIGN UP </Text>
-      </Pressable>
-      
-      
+        <Pressable style={styles.signinbutton} onPress={() => navigation.navigate('Home')}>
+          <Text style={styles.signintext3}> SIGN UP </Text>
+        </Pressable>
 
-    </View>
+        <Pressable
+          style={styles.signinbutton}
+          onPress={async () => {
+            try {
+              // case 1: real phone number, working to send code
+              // const verificationId = await phoneGetConfirmation('+12014076687', recaptchaVerifier);
+
+              // case 2: fake phone number and code from firebase, working
+              const verificationId = await phoneGetConfirmation('+11113334444', recaptchaVerifier);
```

Comment:
[Question] Assuming this is how it works: `phoneGetConfirmation` takes the phone number and recapt ref and returns a temporary verificationId for sign in?